src/ts/address-search.ts

```diff
@@ -1,1240 +1,1441 @@
+
+          // Listen for selection on this item
+          this.picklist.listen(listItem);
+        });
+
+        this.picklist.scrollIntoViewIfNeeded();
+      } else {
+        this.picklist.handleEmptyPicklistLookup(items);
+      }
+
+      // Add a "Powered by Experian" logo to the picklist footer
+      this.poweredByLogo.element = this.poweredByLogo.element || this.poweredByLogo.create(this.picklist);
+
+      // Fire an event after picklist is created
+      this.events.trigger('post-picklist-create', this.picklist.items);
+    };
+
+    this.picklist.handleCommonShowPicklistLogic = () => {
+      // Reset any previously selected current item
+      this.picklist.currentItem = null;
+
+      // Update picklist size
+      this.picklist.size = this.picklist.items?.length;
+
+      // Reset the picklist tab count (used for keyboard navigation)
+      this.picklist.resetTabCount();
+
+      // Hide the inline search spinner
+      this.searchSpinner.hide();
+
+      // Get/Create picklist container element
+      this.picklist.list = this.picklist.list || this.picklist.createList();
+
+      // Ensure previous results are cleared
+      this.picklist.list.innerHTML = '';
+      this.picklist.useAddressEntered.destroy();
+
+      // Fire an event before picklist is created
+      this.events.trigger('pre-picklist-create', this.picklist.items);
+    };
+
+    // Remove the picklist
+    this.picklist.hide = () => {
+      // Clear the current picklist item
+      this.picklist.currentItem = null;
+      // Remove the "use address entered" option too
+      this.picklist.useAddressEntered.destroy();
+      // Remove the "Powered by Experian" logo
+      this.poweredByLogo.destroy(this.picklist);
+
+      if (this.inputs) {
+        // Remove the class denoting a picklist - if Singleline mode is used, then it is the last input field, otherwise use the first one
+        const position = this.searchType === AddressValidationMode.SINGLELINE ? this.inputs.length - 1 : 0;
+        this.inputs[position].classList.remove('showing-suggestions');
+      }
+
+      // Remove the main picklist container
+      if (this.picklist.list) {
+        this.picklist.container.remove();
+        this.picklist.list = undefined;
+      }
+    };
+
+    this.picklist.handleEmptyPicklist = (items: SearchResponse) => {
+      // Create a new item/row in the picklist showing "No matches" that allows the "use address entered" option
+      this.picklist.useAddressEntered.element = this.picklist.useAddressEntered.element || this.picklist.useAddressEntered.create(items.result?.confidence);
+
+      this.picklist.scrollIntoViewIfNeeded();
+
+      // Provide implementing search types with a means of invoking a custom callback
+      if (typeof this.picklist.handleEmptyPicklistCallback === 'function') {
+        this.picklist.handleEmptyPicklistCallback();
+      }
+    };
+
+    this.picklist.handleEmptyWhat3WordsPicklist = (items: LookupW3WResponse) => {

```

Comment:
Is this method (and the one on line 683) the same as the `handleEmptyPicklist` method on line 659? Except from the "items" type? If so, you can use a pipe to make the parameter type more flexible. e.g.

`this.picklist.handleEmptyPicklist = (items: SearchResponse | LookupW3WResponse | LookupV2Response) => {`

Reply:
fixed